3.18.x deps 2023 08 28 16 39 01 610533

larsewi opened the pull request:
- Updated lmdb from 0.9.30 to 0.9.31
- Updated zlib from 1.2.13 to 1.3
- Updated openssl from 1.1.1t to 1.1.1v
- Updated libxml2 from 2.11.2 to 2.11.5
- Updated diffutils from 3.9 to 3.10
- Updated openldap from 2.6.4 to 2.6.6
- Updated libcurl-hub from 8.0.1 to 8.2.1
- Updated git from 2.40.1 to 2.42.0
- Updated postgresql from 13.10 to 13.12
- Updated php from 8.0.28 to 8.0.30
- Updated libcurl from 8.0.1 to 8.2.1

larsewi committed:
Updated lmdb from 0.9.30 to 0.9.31

larsewi committed:
Updated zlib from 1.2.13 to 1.3

larsewi committed:
Updated openssl from 1.1.1t to 1.1.1v

larsewi committed:
Updated libxml2 from 2.11.2 to 2.11.5

larsewi committed:
Updated diffutils from 3.9 to 3.10

larsewi committed:
Updated openldap from 2.6.4 to 2.6.6

larsewi committed:
Updated libcurl-hub from 8.0.1 to 8.2.1

larsewi committed:
Updated git from 2.40.1 to 2.42.0

larsewi committed:
Updated postgresql from 13.10 to 13.12

larsewi committed:
Updated php from 8.0.28 to 8.0.30

larsewi committed:
Updated libcurl from 8.0.1 to 8.2.1

larsewi commented:
@cf-bottom build plz

cf-bottom commented:
Alright, I triggered a build:

[![Build Status](https://ci.cfengine.com//buildStatus/icon?job=pr-pipeline&build=9695)](https://ci.cfengine.com//job/pr-pipeline/9695/)

(with exotics)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/9695/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-9695/

Lex-2008 reviewed (dismissed):
looks goold! Now let's see if it builds.

larsewi committed:
Adjusted cfbuild-openssl.spec for AIX

We need to specify aix-gcc specifically because openssl1.1.1 does not detect it properly.

Ticket: none
Changelog: none

larsewi committed:
Bot Tom forgot to update version number in AIX specs

Updated version number in AIX specs, since Bot Tom forgot to do that.

Ticket: None
Changelog: None
Signed-off-by: Lars Erik Wik <lars.erik.wik@northern.tech>

craigcomstock commented on deps-packaging/openssl/cfbuild-openssl.spec:
```diff
@@ -56,15 +56,22 @@ then
     fi
 fi
 
+AIX_64_BIT_KIT=
```
This is odd, I removed this business and things seems to compile nicely.

I looked at the config script there and it seems like we should be OK but default would be 32 bits for KERNEL_BITS and OBJECT_MODE right? And it would seem that on some build the CC was not gcc but something else... which for us, it should be gcc.

in `/home/jenkins/workspace/testing-pr/label/PACKAGES_ppc64_aix_71/openssl/BUILD/openssl-1.1.1v/config`:

```
785             [ "$KERNEL_BITS" ] || KERNEL_BITS=`(getconf KERNEL_BITMODE) 2>/dev/null`
786             KERNEL_BITS=${KERNEL_BITS:-32}
787             OBJECT_MODE=${OBJECT_MODE:-32}
788             if [ "$CC" = "gcc" ]; then
789                 OUT="aix-gcc"
790               if [ $OBJECT_MODE -eq 64 ]; then
791                 echo 'Your $OBJECT_MODE was found to be set to 64'
792                 OUT="aix64-gcc"
793               fi
794             elif [ $OBJECT_MODE -eq 64 ]; then
795                 echo 'Your $OBJECT_MODE was found to be set to 64'
796                 OUT="aix64-cc"
797             else
798                 OUT="aix-cc"
799                 if [ $KERNEL_BITS -eq 64 ]; then
800                     echo "WARNING! If you wish to build 64-bit kit, then you have to"
801                     echo "         invoke '$THERE/Configure aix64-cc' *manually*."
802                     if [ "$DRYRUN" = "false" -a -t 1 ]; then
803                         echo "         You have ~5 seconds to press Ctrl-C to abort."
804                         (trap "stty `stty -g`; exit 0" 2 0; stty -icanon min 0 time 50; read waste) <&1
805                     fi
806                 fi
807             fi
```

craigcomstock replied:
was it failing sometime later because it was built 32 bit? What was the error that made you introduce this change? :)

craigcomstock commented:
retrying aix builds with reverted aix64 commit, worked two ways manually for me on aix71 so let's see if the job works

craigcomstock commented:
not sure why aix71 is failing or using the odd IBM compiler

```
16:38:05 /opt/IBM/xlc/13.1.3/bin/.orig/cc: 1501-210 (W) command option -O contains an incorrect subargument
16:38:05 ld: 0706-005 Cannot find or open file: no-omit-frame-pointer
16:38:05        ld:fopen(): A file or directory in the path name does not exist.
16:38:05 gmake[1]: Leaving directory '/home/jenkins/workspace/testing-pr/label/PACKAGES_ppc64_aix_71/openssl/BUILD/openssl-1.1.1v'
16:38:05 gmake[1]: *** [Makefile:741: libcrypto.a] Error 255
16:38:05 gmake: *** [Makefile:174: all] Error 2
16:38:05 
16:38:05 
16:38:05 RPM build errors:
```

and if I even just try running that cc I get
```
bash-5.1$ /opt/IBM/xlc/13.1.3/bin/.orig/cc
/opt/IBM/xlc/13.1.3/bin/.orig/cc: 1501-287 (S) This compiler does not support AIX 7.1. Please check with IBM (http://www-01.ibm.com/support/docview.wss?rs=43&uid=swg21326972) to see if there is a PTF for this compiler that supports this AIX level.
```

It should be using /usr/bin/gcc which is /opt/freeware/bin/gcc which is /opt/freeware/bin/gcc-10 (which maybe should be gcc-6 according to setup guidelines we have).

craigcomstock commented:
aix71 retry after fixing that build host configuration/software a bit [![Build Status](https://ci.cfengine.com/buildStatus/icon?job=pr-pipeline&build=9785)](https://ci.cfengine.com/job/pr-pipeline/9785/)

craigcomstock commented:
retry aix71 after making the change that worked for me manually/locally on aix71 build host
[![Build Status](https://ci.cfengine.com/buildStatus/icon?job=pr-pipeline&build=9789)](https://ci.cfengine.com/job/pr-pipeline/9789/)

craigcomstock commented:
@cf-bottom jenkins with exotics please

cf-bottom commented:
Sure, I triggered a build:

[![Build Status](https://ci.cfengine.com//buildStatus/icon?job=pr-pipeline&build=9806)](https://ci.cfengine.com//job/pr-pipeline/9806/)

(with exotics)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/9806/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-9806/

larsewi commented on deps-packaging/openssl/cfbuild-openssl.spec:
```diff
@@ -62,14 +62,17 @@ if [ x$SYS = "xAIX" ]; then
 fi
 
-./config shared  no-idea no-rc5 no-ssl3 no-dtls no-psk no-srp no-engine \
+$PERL ./Configure $target shared  no-idea no-rc5 no-ssl3 no-dtls no-psk no-srp no-engine \
```
RHEL 7 is now broken :( 
```
13:34:14 NOTE: If in doubt, on Unix-ish systems use './config'.
13:34:14 Configuring OpenSSL version 1.1.1v (0x1010116fL) for 
13:34:14 Using os-specific seed configuration
13:34:14 error: Bad exit status from /var/tmp/rpm-tmp.hw4q5q (%build)
```
Maybe we need to use `./config` on none AIX

craigcomstock replied:
yes, working on it now :+1:

craigcomstock commented:
checking rhel7 and aix*, will squash commits when "done" [![Build Status](https://ci.cfengine.com/buildStatus/icon?job=pr-pipeline&build=9816)](https://ci.cfengine.com/job/pr-pipeline/9816/)

craigcomstock commented:
bad shell code in spec file for openssl, retry: [![Build Status](https://ci.cfengine.com/buildStatus/icon?job=pr-pipeline&build=9825)](https://ci.cfengine.com/job/pr-pipeline/9825/)

craigcomstock commented:
@larsewi the CI builds look good, rhel7 🟢 aix5 and aix7 are in progress testing so past the building of openssl that was holding us up.

I squashed the openssl build spec commits into one.

@cf-bottom jenkins please with exotics

cf-bottom commented:
Alright, I triggered a build:

[![Build Status](https://ci.cfengine.com//buildStatus/icon?job=pr-pipeline&build=9828)](https://ci.cfengine.com//job/pr-pipeline/9828/)

(with exotics)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/9828/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-9828/

craigcomstock commented:
> bad shell code in spec file for openssl, retry: [![Build Status](https://camo.githubusercontent.com/05cf85515b955d7a313a77e03764c4503b9befc49c9b23d0836dc9299559f980/68747470733a2f2f63692e6366656e67696e652e636f6d2f6275696c645374617475732f69636f6e3f6a6f623d70722d706970656c696e65266275696c643d39383235)](https://ci.cfengine.com/job/pr-pipeline/9825/)

That build is successful in terms of rhel7, aix5 and aix7 all passing their testing-pr jobs. So @larsewi I think we are nearly ready to go with this PR.

larsewi commented:
@craigcomstock I think we can merge this. Don't you?
